core/core-impl/src/test/java/ai/timefold/solver/core/impl/solver/DefaultSolverTest.java

```diff
@@ -849,6 +887,44 @@ void solveRepeatedlyListVariable(SoftAssertions softly) {
         softly.assertThat(solution.getScore().isSolutionInitialized()).isTrue();
     }
 
+    @Test
+    void solveWithAllowsUnassignedValuesListVariable() {
+        var solverConfig = PlannerTestUtils.buildSolverConfig(TestdataAllowsUnassignedValuesListSolution.class,
+                TestdataAllowsUnassignedValuesListEntity.class, TestdataAllowsUnassignedValuesListValue.class)
+                .withEasyScoreCalculatorClass(TestdataAllowsUnassignedValuesListEasyScoreCalculator.class)
+                .withTerminationConfig(new TerminationConfig().withBestScoreLimit("0"))
+                .withPhases();
+        var solverFactory = SolverFactory.<TestdataAllowsUnassignedValuesListSolution> create(solverConfig);
+        var solver = solverFactory.buildSolver();
+
+        var value1 = new TestdataAllowsUnassignedValuesListValue("v1");
+        var value2 = new TestdataAllowsUnassignedValuesListValue("v2");
+        var value3 = new TestdataAllowsUnassignedValuesListValue("v3");
+        var value4 = new TestdataAllowsUnassignedValuesListValue("v4");
+        var entity = new TestdataAllowsUnassignedValuesListEntity("e1");
+        entity.setValueList(Arrays.asList(value1, value2));
+        value1.setIndex(0);
+        value1.setEntity(entity);
+        value1.setPrevious(null);
+        value1.setNext(value2);
+        value2.setIndex(1);
+        value2.setEntity(entity);
+        value2.setPrevious(value1);
+        value2.setNext(null);
+
+        var solution = new TestdataAllowsUnassignedValuesListSolution();
+        solution.setEntityList(List.of(entity));
+        solution.setValueList(Arrays.asList(value1, value2, value3, value4));
+
+        var bestSolution = solver.solve(solution);
+        assertSoftly(softly -> {
```

Comment:
Construction heuristics only checks unassigned values and therefore will not touch value1 and value2. What was in the lists, will remain in the lists. Local search on the other hand has the power to rearrange the lists, and it will do so. The objective function tells it to prefer unassigned things, so it will unassign every element. 

That's why the list is empty after local search, but not after construction heuristics.